src/Microsoft.AspNetCore.Server.Testing/Deployers/NanoDeployer/NanoDeployer.cs

```diff
@@ -0,0 +1,251 @@
+            {
+                uri = $"http://{_deploymentParameters.ServerName}:80/";
+            }
+            else
+            {
+                uri = $"http://{_deploymentParameters.ServerName}:5000/";
+            }
+
+            return new DeploymentResult
+            {
+                ApplicationBaseUri = uri,
+                DeploymentParameters = DeploymentParameters
+            };
+        }
+
+        public override void Dispose()
+        {
+            try
+            {
+                Logger.LogInformation("Stopping the server...");
+                RunScript("StopServer");
+            }
+            catch { }
+
+            try
+            {
+                Logger.LogInformation($"Deleting the deployed folder '{_deployedFolderPathInFileShare}'");
+                Directory.Delete(_deployedFolderPathInFileShare, recursive: true);
+            }
+            catch { }
+
+            try
+            {
+                Logger.LogInformation($"Deleting the locally published folder '{DeploymentParameters.PublishedApplicationRootPath}'");
+                Directory.Delete(DeploymentParameters.PublishedApplicationRootPath, recursive: true);
+            }
+            catch { }
+        }
+
+        private void RunScript(string serverAction)
+        {
+            var parameterBuilder = new StringBuilder();
+            parameterBuilder.Append($"\"{_remotePSSessionHelperScript}\"");
+            parameterBuilder.Append($" -serverName {_deploymentParameters.ServerName}");
+            parameterBuilder.Append($" -accountName {_deploymentParameters.ServerAccountName}");
+            parameterBuilder.Append($" -accountPassword {_deploymentParameters.ServerAccountPassword}");
+            parameterBuilder.Append($" -executablePath {Path.Combine(_deployedFolderPathInFileShare, _deploymentParameters.NanoServerRelativeExecutablePath)}");
+            parameterBuilder.Append($" -serverType {_deploymentParameters.ServerType}");
+            parameterBuilder.Append($" -serverAction {serverAction}");
+
+            // todo: launch a powershell process to make the website point to the created folder
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = "powershell.exe",
+                Arguments = parameterBuilder.ToString(),
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true
+            };
+
+            using (var deployToNanoProcess = new Process() { StartInfo = startInfo })
+            {
+                deployToNanoProcess.EnableRaisingEvents = true;
+                deployToNanoProcess.ErrorDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogWarning("Nano server: " + dataArgs.Data);
```

Comment:
Can you change the message to be `[Nano server] <text>`? It's much easier to read